juju/juju
# k8s storage is opinionated model config

## Description of change

Juju now uses opinionated storage defaults when provisioning storage on k8s. The main changes:
- add-k8s queries the cluster and prompts for storage if no suitable storage class can be found
- the k8s provider gains config support for storing the workload and operator storage classes as model default attributes
- a bunch of code used to search for k8s storage classes is deleted

## QA steps

Use add-k8s to import microk8s
check that model-defaults shows hostpath storage for operator and workload
add a model
check that model config shows hostpath storage for operator and workload
deploy a charm with storage
create a storage pool and deploy a charm with pool storage
